monitoring/resourse_lakehouse_monitor.go

```diff
@@ -0,0 +1,139 @@
+package monitoring
+
+import (
+	"context"
+
+	"github.com/databricks/databricks-sdk-go/service/catalog"
+	"github.com/databricks/terraform-provider-databricks/common"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+)
+
+func removeSnapshotField(originalSchema map[string]*schema.Schema) map[string]*schema.Schema {
+	tmpSchema := make(map[string]*schema.Schema)
+	for k, v := range originalSchema {
+		tmpSchema[k] = v
+	}
+	delete(tmpSchema, "snapshot")
+	return tmpSchema
+}
+
+type Snapshot struct {
+}
+type Monitor struct {
```

Comment:
Why do we need this struct? Why can't we use `catalog.MonitorInfo` directly?

Reply:
We need this struct because of the `Snapshot` field. Snapshot is of type `any` and it is not compatible with the schema. Therefore this custom struct uses an empty struct for snapshot, and then initializes the snapshot field later in the create and update below.

Reply:
Why is `Snapshot` `any`? What is its actual type? Is it mean to be `struct{}`, and this is a bug with the types in the Go SDK? Or can users actually pass any value for this field?